### ISSUE TITLE
[@angular-eslint/require-input-non-null-assertion] Enforce ! Modifier for Required @Input() Properties

### ISSUE THREAD

**Issue by Dmytro27Ind** [angular-eslint/angular-eslint#2163]
In Angular, when using `@Input()` with `{ required: true }`, it's essential to ensure that the associated property is marked with the non-null assertion operator (`!`) to comply with TypeScript's strict property initialization rules. However, there is currently no ESLint rule in `@angular-eslint` to enforce this practice.

### Example of the Problem

The following code does not use the `!` modifier, which may lead to initialization issues:

```ts
@Input({ required: true })
value: number;
```

The correct version should enforce the `!` modifier as follows:

```ts
@Input({ required: true })
value!: number;
```

### Proposed Solution

Introduce a new rule, e.g., `@angular-eslint/require-input-non-null-assertion`, which validates that:

1. All `@Input()` properties marked with `{ required: true }` use the `!` modifier.
2. Optionally, the rule could allow configuration to enforce or ignore specific cases (e.g., `string | null`).

### Benefits

- Improves consistency across Angular projects.
- Encourages adherence to TypeScript's `strictPropertyInitialization` rules.

### Example Rule Behavior

| Code                                   | Pass/Fail |
|----------------------------------------|-----------|
| `@Input({ required: true }) value!: number;` | ✅ Pass    |
| `@Input({ required: true }) value: number;`  | ❌ Fail    |
| `@Input() value: number;`               | ✅ Pass    |

**Comment by reduckted** [comment]
> Encourages adherence to TypeScript's strictPropertyInitialization rules.

What purpose would this rule serve when enabling the `strictPropertyInitialization` compiler option would achieve the same thing?

**Comment by Dmytro27Ind** [comment]
@reduckted Hi. I have added new examples in the description of this issue.

The main reason why it is better to create a separate rule is that if you use `strictPropertyInitialization`  , TS will allow you to do it this way:

`@Input({ required: true }) value: number = 2;`

But it doesn't make sense because `default value` will never be used, because `@Input` is **required**.

**Comment by reduckted** [comment]
Oh, that makes more sense now. :+1:

It's worth noting that a rule like this would end up being redundant if `prefer-signals` is used with the `preferInputSignals` option (which is true by default).
